libraries/AP_HAL_ChibiOS/Scheduler.cpp

```diff
@@ -472,4 +480,24 @@ bool Scheduler::thread_create(AP_HAL::MemberProc proc, const char *name, uint32_
     return true;
 }
 
+/*
+  inform the scheduler that we are calling an operation from the
+  main thread that may take an extended amount of time. This can
+  be used to prevent watchdog reset during expected long delays
+  A value of zero cancels the previous expected delay
+*/
+void Scheduler::expect_delay_ms(uint32_t ms)
+{
+    if (!in_main_thread()) {
+        // only for main thread
+        return;
+    }
+    if (ms == 0) {
+        expect_delay_start = 0;
```

Comment:
`expect_delay_length` should be reset here, or you get a extra round of pats on time wrap.

Reply:
i don't think that's right. It checks for zero.